log/stat/stat.go

```diff
@@ -0,0 +1,97 @@
+package stat
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/peak/s5cmd/strutil"
+)
+
+const (
+	totalCount = iota
+	succCount
+	timeInfo
+)
+
+var (
+	enabled bool
+	stats   statistics
+)
+
+type statistics [3]syncMap
+
+// InitStat initializes collecting program statistics.
+func InitStat() {
+	enabled = true
+	for i := range stats {
+		stats[i] = syncMap{
+			mu:          sync.Mutex{},
+			mapStrInt64: map[string]int64{},
+		}
+	}
+}
+
+// syncMap is a statically typed and synchronized map.
+type syncMap struct {
+	mu          sync.Mutex
+	mapStrInt64 map[string]int64
+}
+
+func (s *syncMap) add(key string, val int64) {
+	s.mu.Lock()
+	s.mapStrInt64[key] += val
+	s.mu.Unlock()
+}
+
+// Stat implements log.Message interface.
+type Stat struct {
+	Visited     string  `json:"visited"`
+	SuccVisits  int64   `json:"success visits"`
+	ErrVisits   int64   `json:"error visits"`
+	AvgExecTime float64 `json:"avg. execution time"`
+}
+
+func (s Stat) String() string {
```

Comment:
It is better to have more readable stats output.
Current implementation dumps stats as below:
```
"newSession" 1 time with 0 ERROR and 1 SUCCESS returns; average execution time: 0.000000 msec.
"S3.Get" 11 time with 0 ERROR and 11 SUCCESS returns; average execution time: 591.909091 msec.
"Copy.doDownload" 11 time with 0 ERROR and 11 SUCCESS returns; average execution time: 592.545455 msec.
"Copy.Run" 1 time with 0 ERROR and 1 SUCCESS returns; average execution time: 1492.000000 msec.
```

I am suggesting the below tabular format: 
```
Operation        Total Error Success  Duration(ms)
==================================================
newSession           1     0       1             0
S3.Get              11     0      11           591
Copy.doDownload     11     0      11           592
Copy.Run             1     0       1          1492
```
Since this is for us (humans) it should be as readable/simple as it can. What do you think?

Reply:
thanks, tabular format is certainly more pleasant and better